core/rpc/src/state/mod.rs

```diff
@@ -298,17 +304,24 @@ impl<B, E, Block: BlockT, RA> State<B, E, Block, RA> where
 		let mut changes_map: BTreeMap<NumberFor<Block>, StorageChangeSet<Block::Hash>> = BTreeMap::new();
 		for key in keys {
 			let mut last_block = None;
-			for (block, _) in self.client.key_changes(begin, end, key)? {
+			let mut last_value = last_values.get(key).cloned().unwrap_or_default();
```

Comment:
Do we have to clone here? Can't we compare with `value_at_block.as_ref()`?

Reply:
It is also used as storage for the last value (it is updated in the loop). I could use the same `last_values` map for this, but it is single `clone()` call vs multiple `HashMap::get()` + `HashMap::insert()` calls. Could change that if you feel this is better.